Initial Product list UI for cell

syedtazaidi opened the pull request.

syedtazaidi committed:
ProductListUI

Add elements inside ProductTableViewCell

Add custom StarRatingsView for ratings

Add sample code inside ProductListViewController

syedtazaidi committed:
Add productListUI

Add productListUI

designerGenes commented on CapitalOneShoppingCartDemo/CapitalOneShoppingCartDemo/ProductList/StarRatingsView.swift:
```diff
@@ -0,0 +1,47 @@
+/// A view that displays a star rating.
+public class StarsView: UIView {
+  /// The rating to display.
+  public var rating: Int = 0 {
```
this implementation code probably belongs in the presenter

syedtazaidi replied:
Isn't this purely related to StarsView to show images based on the rating?  This is the property we need to assign the actual rating which will be coming from the network. THis is the just initialization within the View.  Let me know if does not make sense

designerGenes replied:
I thought I was still looking at the view controller.  This should be fine

syedtazaidi replied:
@designerGenes  @RituIosDev  @prash-mobiledev  I think we should merge this and all code for 2nd screen today before going forward . For now, this is the UI I'm getting. I will make it cleaner once we have all the Data Model setup once the Data persistence logic is merged. 

<img width="349" alt="image" src="https://user-images.githubusercontent.com/104391800/166694349-3ef9a552-f2fd-41f5-9d4f-2a4e66e42351.png">

designerGenes reviewed (approved):
needs a presenter.  we can handle that on another branch if you'd like though

syedtazaidi committed:
Merge branch 'ProductListScreen' into productListUI

syedtazaidi committed:
Update methods inside presenter

Update methods inside presenter

syedtazaidi committed:
Product List Cell

Add UIImage loading class
Get Image and show placeholder image
Add Button for add to cart button
Add stock label button
Add publisher in the ProductListPresenter
add default product get in the Product Entity

syedtazaidi committed:
Update ProductListViewController.swift

Add ratings in the Cell logic

prash-mobiledev reviewed (approved):
Looks good
